packages/backend/server/src/core/auth/guard.ts

```diff
@@ -36,6 +40,26 @@ export class AuthGuard implements CanActivate, OnModuleInit {
 
   async canActivate(context: ExecutionContext) {
     const { req, res } = getRequestResponseFromContext(context);
+    // rpc request is internal
+    const isInternal = this.reflector.getAllAndOverride<boolean>(
+      INTERNAL_ENTRYPOINT_SYMBOL,
+      [context.getClass(), context.getHandler()]
+    );
+    if (isInternal) {
+      if (req.get('CF-Connecting-IP')) {
```

Comment:
this makes no sense because selfhost user may deploy services onebyone but not under CF gateway